### PR TITLE
[Compression - 7] Add compression metadata

### PR DESCRIPTION
**Pull request by piraka9011**
## Changes
* Bump metadata version (2 -> 3)
* Adds fields:
  * `compression_format`
  * `compression_mode`
* Update tests to account for new metadata info
* **metadata version 2 is still supported**
  * Add unit test to verify metadata with version 2 can still be read.

### CODE REVIEW

File: rosbag2/test/rosbag2/test_info.cpp [hunk 1]
```diff
@@ -53,7 +121,9 @@ TEST_F(TemporaryDirectoryFixture, read_metadata_makes_appropriate_call_to_metada
     "        name: topic2\n"
     "        type: type2\n"
     "        serialization_format: rmw2\n"
-    "      message_count: 200");
+    "      message_count: 200\n"
+    "  compression_format: \"zstd\"\n"
+    "  compression_mode: \"file\"\n");
```
Comment (piraka9011) [inline rosbag2/test/rosbag2/test_info.cpp]: This should be `FILE` (all caps)

File: rosbag2_tests/resources/cdr_test/metadata.yaml [hunk 2]
```diff
@@ -19,3 +19,5 @@ rosbag2_bagfile_information:
         type: test_msgs/Arrays
         serialization_format: cdr
       message_count: 4
+  compression_format: ""
```
Comment (piraka9011) [inline rosbag2_tests/resources/cdr_test/metadata.yaml]: Do these need to be populated? Or at least set to `NONE`?

**Comment by zmichaels11** [comment]
@ros2/aws-oncall - please run this CI job
Gist: https://gist.githubusercontent.com/zmichaels11/071d3dfa2ab96193b88add6db952218a/raw/d79e3aea2a41516610cada96ba65eb2dca99b076/ros2.repos
BUILD args: --packages-up-to rosbag2 rosbag2_storage rosbag2_storage_default_plugins rosbag2_tests
TEST args: --packages-select rosbag2 rosbag2_storage rosbag2_storage_default_plugins rosbag2_tests
Job: ci_launcher

**Comment by dabonnie** [comment]
> --packages-up-to rosbag2 rosbag2_storage rosbag2_storage_default_plugins rosbag2_tests

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8958/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4728/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7330/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8876/)

**Comment by zmichaels11** [comment]
It looks like Windows failed due to buildfarm issues.
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/)

**Comment by piraka9011** [comment]
@Karsten1987 The windows build farm has been super flakey, so we compiled and tested on our bare metal windows machine successfully.

I'm running one more CI, however, Windows build does pass locally.

* Windows [Build Status](https://ci.ros2.org/job/ci_windows/8886)

**Pull request merged**